sdk/core/azure-core/inc/azure/core/internal/io/filehandleholder.hpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief An internal FILE* Holder.
+ *
+ */
+#pragma once
+
+#include <cstdio>
+
+namespace Azure { namespace IO { namespace Internal {
+
+  /**
+   * @brief FileHandleHolder ensures native handle resource is released.
+   */
+  class FileHandleHolder {
+  private:
+    FILE* m_value = nullptr;
+
+  public:
+    explicit FileHandleHolder() noexcept : m_value(nullptr){};
```

Comment:
Why do we want to support creating an empty FileHandleHolder with nothing on it?
I would recommend deleting this cttor and making this class to be constructed by constructor and not by copy operator

Reply:
Will do, was following precedent with the hkey holder coding pattern.